src/dataflow/analysis/flow-graph.ts

```diff
@@ -201,6 +292,12 @@ export class BackwardsPath {
   get endEdge(): Edge {
     return this.edges[this.edges.length - 1];
   }
+
+  checkFailureString(check: string) : string {
+    const edgesInPath = this.edges.slice().reverse();
+    const pathString = edgesInPath.map(e => e.label).join(' -> ');
+    return (`${check} failed for path: ${pathString}`);
```

Comment:
This probably shouldn't live here. BackwardsPath doesn't really have anything to do with checks, it's really just a series of edges in the graph. Maybe a method like `BackwardsPath.toString()` would make sense though (doing just the ' -> ' concatenation part of this)?

Reply:
Good point. Done.